core/domain/classifier_domain_test.py

```diff
@@ -161,3 +161,147 @@ def test_validation(self):
         with self.assertRaisesRegexp(utils.ValidationError, (
             'Expected cached_classifier_data to be a dict')):
             classifier.validate()
+
+
+class TrainClassifierJobDomainTests(test_utils.GenericTestBase):
+    """Test the TrainClassifierJob domain."""
+
+    def test_to_dict(self):
+        expected_training_job_dict = {
+            'job_id': 'exp_id1.SOME_RANDOM_STRING',
+            'algorithm_id': "LDAStringClassifier",
+            'exp_id': 'exp_id1',
+            'exp_version_when_created': 1,
+            'state_name': 'a state name',
+            'status': 'PENDING',
+            'training_data': [
+                {
+                    'answer_group_index': 1,
+                    'answers': ['a1', 'a2']
+                },
+                {
+                    'answer_group_index': 2,
+                    'answers': ['a2', 'a3']
+                }
+            ]
+        }
+        observed_training_job = classifier_domain.TrainClassifierJob(
+            expected_training_job_dict['job_id'],
+            expected_training_job_dict['algorithm_id'],
+            expected_training_job_dict['exp_id'],
+            expected_training_job_dict['exp_version_when_created'],
+            expected_training_job_dict['state_name'],
+            expected_training_job_dict['status'],
+            expected_training_job_dict['training_data'])
+        self.assertDictEqual(expected_training_job_dict,
+                             observed_training_job.to_dict())
+
+    def test_validation(self):
+        """Tests to verify validate method of TrainClassifierJob domain."""
+
+        # Verify no errors are raised for correct data.
+        training_data = [
+            {
+                'answer_group_index': 1,
+                'answers': ['a1', 'a2']
+            },
+            {
+                'answer_group_index': 2,
+                'answers': ['a2', 'a3']
+            }
+        ]
+        training_job_dict = {
+            'job_id': 'exp_id1.SOME_RANDOM_STRING',
+            'exp_id': 'exp_id1',
+            'exp_version_when_created': 1,
+            'state_name': 'some state',
+            'algorithm_id': "LDAStringClassifier",
+            'training_data': training_data,
+            'status': 'PENDING'
+        }
+        training_job = classifier_domain.TrainClassifierJob(
```

Comment:
It might be worth factoring this call out into its own private method within this test class.